#ifndef GUI_ADD_STORAGE_COSTS_DLG_H
#define GUI_ADD_STORAGE_COSTS_DLG_H

////////////////////////////////////////////////////////////////////////////////
//
// Includes
//
#ifndef FM_CORE_H
#	include "fm_core.h"
#endif

// Ui
#include "ui_addstoragecosts.h"

// Qt Includes
#include <QDialog>
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
namespace fm {
////////////////////////////////////////////////////////////////////////////////
namespace gui {
////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////
//
// class CAddStorageCostsDlg
//
class CAddStorageCostsDlg : public QDialog
{
	Q_OBJECT

public:// Constructors
	CAddStorageCostsDlg(QWidget* pwParent = nullptr);
	~CAddStorageCostsDlg() = default;

public:
	//
	//	Main Interface
	//
	// Customer Data Accessors
	inline double  GetCosts() const;
	inline QString GetInfoText() const;
	// Clear Dialog
	inline void Clear();

	public slots:
	void OnAdd();

protected:// Helper Methodes
	void SetLabelColor(QLabel* pLabel, bool bValid);

private:
	//
	//	Content
	//
	//Ui::addStorageCostsDlg m_uiStorageCostsDlg;
	Ui::addStorageCostsDlg m_uiAddStorageCostsDlg;
};
////////////////////////////////////////////////////////////////////////////////



////////////////////////////////////////////////////////////////////////////////
//
//	Inline Implementations
//

inline double CAddStorageCostsDlg::GetCosts() const
{
	QString strCosts = m_uiAddStorageCostsDlg.editCosts->displayText();
	return strCosts.toDouble();
}

inline QString CAddStorageCostsDlg::GetInfoText() const
{
	QString strCosts = m_uiAddStorageCostsDlg.editInfoText->displayText();
	return strCosts;
}

// Clear
inline void CAddStorageCostsDlg::Clear()
{
	m_uiAddStorageCostsDlg.editCosts->clear();

	SetLabelColor(m_uiAddStorageCostsDlg.lblCosts, true);
}

////////////////////////////////////////////////////////////////////////////////
} // namespace gui
////////////////////////////////////////////////////////////////////////////////
} // namespace fm
////////////////////////////////////////////////////////////////////////////////

#endif // GUI_ADD_STORAGE_COSTS_DLG_H